dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataOwnersManagerTest.kt

```diff
@@ -76,4 +92,58 @@ class DataOwnersManagerTest {
         }
         assertTrue(exception.summary.contains("User is already a data owner for company."))
     }
+
+    @Test
+    fun `check that email for users becoming company data owner is not generated if company does not exist`() {
+        val mockStoredCompany = mock(StoredCompanyEntity::class.java)
+
+        `when`(mockStoredCompany.companyName).thenReturn(testCompanyName)
+        `when`(mockStoredCompany.companyId).thenReturn(UUID.randomUUID().toString())
+
+        val exception = assertThrows<ResourceNotFoundApiException> {
+            dataOwnersManager.addDataOwnerToCompany(
+                companyId = mockStoredCompany.companyId,
+                userId = testUserId,
+                companyName = mockStoredCompany.companyName,
+            )
+        }
+        assertTrue(exception.summary.contains("Company is invalid"))
+    }
+
+    @Test
+    fun `check that email generated for users becoming company data owner are generated`() {
+        val mockStoredCompany = mock(StoredCompanyEntity::class.java)
+        val companyId = UUID.randomUUID().toString()
+
+        `when`(mockStoredCompany.companyName).thenReturn(testCompanyName)
+        `when`(mockStoredCompany.companyId).thenReturn(companyId)
```

Comment:
same as above (mockStoredCompany really needed?):
mockStoredComany.companyName --> companyName
mockStoredCompany.companyId --> companyId

Reply:
Also right here. It is revised.